apps/vscode/command_client.py

```diff
@@ -210,10 +222,14 @@ def read_json_with_timeout(path: str) -> Any:
             pass
 
         actions.sleep(sleep_time)
-        sleep_time *= 2
-        if time.perf_counter() - start_time > VSCODE_COMMAND_TIMEOUT:
+
+        time_left = timeout_time - time.perf_counter()
+
+        if time_left < 0:
             raise Exception("Timed out waiting for response")
 
+        sleep_time = min(sleep_time * 2, time_left)
```

Comment:
You should clamp the minimum sleep time as per my example as a precaution, because if there's just the right amount of slip between the perf_counter clock and the actual sleep() time, you could end up looping extra times with very small sleeps.
e.g.
`sleep_time = max(min(sleep_time * 2, time_left), INITIAL_SLEEP_TIME_SECONDS)`